engine/src/main/java/io/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -237,10 +240,12 @@ public void afterExecutionPathCompleted(
 
         final var item = inputCollection.get(loopCounter);
         createInnerInstance(element, flowScopeContext, item);
+
+        childInstanceCreated = true;
       }
     }
 
-    if (stateBehavior.canBeCompleted(childContext)) {
+    if (!childInstanceCreated && stateBehavior.canBeCompleted(childContext)) {
```

Comment:
Please add a comment why the `canBeCompleted` is not enough. While reading the following question pops up: if it did create a child, then why doesn't the `canBeCompleted` just return false.